src/main/java/com/atlassian/db/replica/internal/RouteDecisionBuilder.java

```diff
@@ -0,0 +1,64 @@
+package com.atlassian.db.replica.internal;
+
+import com.atlassian.db.replica.api.context.Reason;
+import com.atlassian.db.replica.api.context.RouteDecision;
+
+import java.util.Objects;
+
+public final class RouteDecisionBuilder {
+    private String sql;
+    private Reason reason;
+    private RouteDecision currentStateDecision;
```

Comment:
As a fan of short names, I gotta try:
```suggestion
    private RouteDecision cause;
```
😉